api/posts/getPosts.ts

```diff
@@ -0,0 +1,10 @@
+import 'server-only'
+
+import { cache } from 'react'
+
+import { client } from '../../sanity/lib/client'
+
+export const getPosts = cache(async () => {
+  const posts = await client.fetch('*[_type == "post"]')
+  return posts
+})
```

Comment:
Since the Sanity `client` is maintained by the `@sanity/client` package, I'm not sure it uses `fetch` under the hood. If that is true, we don't get access to Next's ability to dedupe requests automatically. The Next docs instruct us to use React's `cache()` fn to enable deduping.

https://nextjs.org/docs/app/building-your-application/data-fetching/caching#react-cache

If someone knows more about the `@sanity/client` implementation and thinks this isn't needed, lmk.

Reply:
how i've seen it done in a template I worked on was using the `next-sanity` library which uses `cache()` under the hood - https://github.com/daltn/lauren-brady/blob/main/lib/sanity.client.ts https://github.com/sanity-io/next-sanity

Reply:
`cache` seems to be the way to go here